crates/nu-path/src/tilde.rs

```diff
@@ -60,59 +75,47 @@ fn user_home_dir(username: &str) -> PathBuf {
     // Returns home dir of user.
 }
 
-#[cfg(target_os = "macos")]
+#[cfg(any(target_os = "android", target_os = "windows", target_os = "macos"))]
 fn user_home_dir(username: &str) -> PathBuf {
+    use std::path::Component;
+
+    let fallback = || {
+        let mut expected_path = String::from(FALLBACK_USER_HOME_BASE_DIR);
+        expected_path.push_str(username);
+        let path = Path::new(&expected_path);
+        let mut home = PathBuf::new();
+        home.push(path);
+        home
+    };
+
     match dirs_next::home_dir() {
         None => {
-            let mut expected_path = String::from("/Users/");
-            expected_path.push_str(username);
-            let path = Path::new(&expected_path);
-            let mut home = PathBuf::new();
-            home.push(path);
-            home
-        }
-        Some(user) => {
-            let mut expected_path = user;
-            expected_path.pop();
-            expected_path.push(Path::new(username));
-            if expected_path.is_dir() {
-                expected_path
-            } else {
-                let mut expected_path_as_string = String::from("/Users/");
-                expected_path_as_string.push_str(username);
-                let path = Path::new(&expected_path_as_string);
-                let mut home = PathBuf::new();
-                home.push(path);
-                home
+            // Termux always has the same home directory
+            #[cfg(target_os = "android")]
+            if std::env::var("TERMUX_VERSION").is_ok() {
```

Comment:
I'm not sure if this should be allowed. We usually prohibit accessing the environment variable except through `stack.get_env_var()`, with the exception of testing.

Reply:
Hm, ok, what do you suggest? It doesn't look like the stack or engine state are accessible in this context, nor in many of the upstream call sites, and fixing that might be a larger change.

Apologies, I'm not very familiar with nushell internals; what is the consequence of getting an environment variable directly from the process, rather than through the nushell engine?

Reply:
I think in this instance, it is just checking whether Nushell was launched from the Termux terminal emulator, right? In that case, it should be fine, but I'd wrap it in something like `is_termux_terminal()` with a comment to denote what it's supposed to mean.

Reply:
That's right, it's just checking that we're actually running the Termux app, rather than just Android in some other context.

I've made the suggested change.